devicetypes/smartthings/zigbee-battery-accessory-dimmer.src/zigbee-battery-accessory-dimmer.groovy

```diff
@@ -118,9 +126,50 @@ def handleSengledSwitchEvents(descMap) {
 	}
 }
 
+def handleCentraliteSmartSwitchEvents(descMap) {
+	if (descMap.clusterInt == 0x0006) {
+		if (descMap.commandInt == 0x01) {
+			sendEvent(name: "switch", value: "on")
+		} else if (descMap.commandInt == 0x00) {
+			sendEvent(name: "switch", value: "off")
+		}
+	} else if (descMap.clusterInt == 0x0008) {
+		def currentLevel = device.currentValue("level") as Integer ?: 0
+		if (descMap.commandInt == 0x05) {
+			def value = Math.min(currentLevel + DOUBLE_STEP, 100)
+			log.debug "move up"
+			// don't change level if switch will be turning off
+			sendEvent(name: "level", value: value)
+			sendEvent(name: "switch", value: "on")
+		} else if (descMap.commandInt == 0x01) {
+			log.debug "move down"
+			def value = Math.max(currentLevel - DOUBLE_STEP, 0)
+			if(value == 0) {
+				sendEvent(name: "switch", value: "off" )
+			} else {
+				sendEvent(name: "switch", value: "on")
+				sendEvent(name: "level", value: value)
+			}
+		}
+	}
+}
+
 def handleBatteryEvents(descMap) {
+	def batteryValue = null
 	if (descMap?.clusterInt == zigbee.POWER_CONFIGURATION_CLUSTER && descMap?.value) {
-		def batteryValue = zigbee.convertHexToInt(descMap.value) / 2
+		if(isCentraliteSwitch()){
+			if(descMap?.value) {
```

Comment:
You have a check above for descMap.value, so this is redundant. Also, in the if statement above, you only need the `descMap?.` in the first check. You can safely reference descMap.value because if descMap is null it will fail the first check with `descMap?.`.

Reply:
Sure